pkg/client/macros.go

```diff
@@ -15,22 +18,68 @@ func (h *V1Client) CreateMacros(uid string, macros *models.V1Macros) error {
 	}
 	if uid != "" {
 		params := userC.NewV1ProjectsUIDMacrosCreateParams().WithContext(h.Ctx).WithUID(uid).WithBody(macros)
-		_, err := client.V1ProjectsUIDMacrosCreate(params)
-		if err != nil {
-			return err
-		}
+		_, err = client.V1ProjectsUIDMacrosCreate(params)
 	} else {
-		tenantUID, err := h.GetTenantUID()
+		var tenantUID string
+		tenantUID, err = h.GetTenantUID()
 		if err != nil {
 			return err
 		}
 		params := userC.NewV1TenantsUIDMacrosCreateParams().WithContext(h.Ctx).WithTenantUID(tenantUID).WithBody(macros)
 		_, err = client.V1TenantsUIDMacrosCreate(params)
+	}
+	if err != nil {
+		err = h.handleMacroDuplicateForbiddenError(macros, uid, err)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (h *V1Client) handleMacroDuplicateForbiddenError(macros *models.V1Macros, projectUID string, srcErr error) error {
+	if strings.Contains(srcErr.Error(), "Code:DuplicateMacroNamesForbidden") {
+		var srcMacro *models.V1Macro
+		srcMacro = macros.Macros[0]
+		actualMacro, err := h.retryGetMacro(3, 10*time.Second, srcMacro.Name, projectUID)
 		if err != nil {
 			return err
 		}
+		if actualMacro != nil {
+			if (srcMacro.Name == actualMacro.Name) && (srcMacro.Value == actualMacro.Value) {
+				return nil
+			} else if (srcMacro.Name == actualMacro.Name) && (srcMacro.Value != actualMacro.Value) {
+				return srcErr
+			}
+		} else {
+			return errors.New("code: Failed to create macro")
+		}
+	} else {
+		return srcErr
 	}
+	return nil
+}
+func (h *V1Client) retryGetMacro(attempts int, sleep time.Duration, macroName string, projectUid string) (*models.V1Macro, error) {
```

Comment:
@SivaanandM could you please use go-retry library here? Let's make retry configurable with exponential backoff.